plugin/src/main/java/org/opensearch/ml/action/tasks/GetTaskTransportAction.java

```diff
@@ -84,6 +141,111 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLTask
+                        connectorExecutor.setScriptService(scriptService);
+                        connectorExecutor.setClusterService(clusterService);
+                        connectorExecutor.setClient(client);
+                        connectorExecutor.setXContentRegistry(xContentRegistry);
+                        connectorExecutor.executeAction(BATCH_STATUS.name(), mlInput, ActionListener.wrap(taskResponse -> {
+                            processTaskResponse(mlTask, taskId, taskResponse, transformJob, actionListener);
+                        }, e -> { actionListener.onFailure(e); }));
+                    }, e -> {
+                        log.error("Failed to get connector " + model.getConnectorId(), e);
+                        actionListener.onFailure(e);
+                    });
+                    try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
+                        connectorAccessControlHelper
+                            .getConnector(client, model.getConnectorId(), ActionListener.runBefore(listener, threadContext::restore));
+                    }
+                } else {
+                    actionListener.onFailure(new ResourceNotFoundException("Can't find connector " + model.getConnectorId()));
+                }
+            }, e -> {
+                log.error("Failed to retrieve the ML model with the given ID", e);
+                actionListener.onFailure(e);
+            }));
+        } catch (Exception e) {
+            // fetch the connector
+            log.error("Unable to fetch status for ml task ", e);
+        }
+    }
+
+    private void processTaskResponse(
+        MLTask mlTask,
+        String taskId,
+        MLTaskResponse taskResponse,
+        Map<String, Object> transformJob,
+        ActionListener<MLTaskGetResponse> actionListener
+    ) {
+        try {
+            ModelTensorOutput tensorOutput = (ModelTensorOutput) taskResponse.getOutput();
+            if (tensorOutput != null && tensorOutput.getMlModelOutputs() != null && !tensorOutput.getMlModelOutputs().isEmpty()) {
+                ModelTensors modelOutput = tensorOutput.getMlModelOutputs().get(0);
+                if (modelOutput.getMlModelTensors() != null && !modelOutput.getMlModelTensors().isEmpty()) {
+                    Map<String, Object> transformJobStatus = (Map<String, Object>) modelOutput.getMlModelTensors().get(0).getDataAsMap();
+                    if (transformJobStatus != null) {
+                        transformJob.putAll(transformJobStatus);
+                        Map<String, Object> updatedTask = new HashMap<>();
+                        updatedTask.put(TRANSFORM_JOB_FIELD, transformJob);
+
+                        if ((transformJob.containsKey("status") && transformJob.get("status").equals("completed"))
+                            || (transformJob.containsKey("TransformJobStatus")
```

Comment:
Such parsing logic looks only limited to Sagemaker, right ? Does this PR only support Sagemaker ? Suggest add some comment

Reply:
Its both sagemaker and OpenAI. Will add more comments